addons/web/static/src/views/fields/input_field_hook.js

```diff
@@ -1,5 +1,6 @@
 /** @odoo-module **/
```

Comment:
???? [REF] web: earlier record update for dirty field

A field input always asks for a record update when it is the target of a "change" event
but such an event might occur sometimes before some keyboard events (e.g. a "Enter"). In order to make
keyboard navigation work, it is necessary to make a dirty field to communicate its new value before a change event and make sure the record is correctly updated before being saved.

Before this commit, this was done via a system based on a special event triggered on the model each time a record is
saved but that solution was not appropriate:

- the updates done in that way were not correctly awaited.
- they would be asked for even if the save would not be the result of a keyboard event.

Here we replace the old system by a simpler one that works. When some specific keyboard events occur, a dirty field
ask for an update.
 



Before this commit, when a record would be saved, it would ask to all its related fields
to communicate any input change that would be not known by it, that is ask for an update.
Normally, an update is asked whenever a field input is the targe of a "change" event
Unfortunately, the udpates made that way were not awaited so the record could be
saved without the change.

The purpose of this commit is to not have to tell the fields to apply
their update when saving. The only known way to save a record without
applying updates to the current fields is to apply updates to the keynav.
We will apply an update when we do a tab/shift+tab/enter.

Reply:
occur **after** some keyboard